user/user_test.go

```diff
@@ -40,31 +43,362 @@ func TestRESTClient_NewUser(t *testing.T) {
+
+func TestRESTClient_UpdateUser_EmptyUserName(t *testing.T) {
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	u := &model.User{Username: ""}
+
+	err := cl.UpdateUser(ctx, u)
+
+	assert.Error(t, err)
+
+	p.AssertExpectations(t)
+}
+
+func TestRESTClient_UpdateUser_UserNotProvided(t *testing.T) {
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	u := &model.User{}
+	u = nil
+
+	err := cl.UpdateUser(ctx, u)
+
+	assert.Error(t, err)
+
+	p.AssertExpectations(t)
+}
+
+func TestRESTClient_DeleteUser(t *testing.T) {
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	u := &model.User{
+		Username: exampleUser,
+	}
+
+	deleteParams := &products.DeleteUsersUserIDParams{
+		UserID:  u.UserID,
+		Context: ctx,
+	}
+
+	getParams := &products.GetUsersParams{
+		Context:  ctx,
+		Username: &u.Username}
+
+	p.On("GetUsers", getParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+		Return(&products.GetUsersOK{
+			Payload: []*model.User{{Username: exampleUser}},
+		}, nil)
+
+	p.On("DeleteUsersUserID", deleteParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+		Return(&products.DeleteUsersUserIDOK{}, nil)
+
+	err := cl.DeleteUser(ctx, u)
+
+	assert.NoError(t, err)
+
+	p.AssertExpectations(t)
+}
+
+func TestRESTClient_DeleteUser_EmptyUserName(t *testing.T) {
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	u := &model.User{Username: ""}
+
+	err := cl.DeleteUser(ctx, u)
+
+	assert.Error(t, err)
+
+	p.AssertExpectations(t)
+}
+
+func TestRESTClient_DeleteUser_UserNotProvided(t *testing.T) {
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	u := &model.User{}
+	u = nil
+
+	err := cl.DeleteUser(ctx, u)
+
+	assert.Error(t, err)
+
+	p.AssertExpectations(t)
+}
+
+func TestRESTClient_DeleteUser_UserIDMismatch(t *testing.T) {
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	u := &model.User{
+		Username: exampleUser,
+		UserID:   1,
+	}
+
+	u2 := &model.User{
+		Username: exampleUser,
+		UserID:   2,
+	}
+
+	getParams := &products.GetUsersParams{
+		Context:  ctx,
+		Username: &u.Username,
+	}
+
+	p.On("GetUsers", getParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+		Return(&products.GetUsersOK{
+			Payload: []*model.User{u},
+		}, &ErrUserMismatch{})
+
+	err := cl.DeleteUser(ctx, u2)
+
+	assert.Equal(t, err, &ErrUserMismatch{})
```

Comment:
You're doing a type check here right? Testify package provides a method `assert.IsType()` for this